### PR TITLE
✨ add possibility to pass history in constructor

### PR DESCRIPTION
**Pull request by garak**
Main feature is described in title.

Some other optimizations:
* remove `history` from main class, since it's only used in PgnOutput (added there with a different name)
* make History and Move stringable
* add the possibility to use an identifier in HtmlOutput
* fix some tests using `assertSame` with swapped arguments

### CODE REVIEW

**Comment by garak** [comment]
Notice that Chess class has improved coverage (now **hundred percent**!) and reduced complexity 😉

File: src/Chess.php [hunk 1]
```diff
@@ -31,12 +31,18 @@ class Chess
...
+     * @param array<int, History>|null $history
+     */
+    public function __construct(?string $fen = Board::DEFAULT_POSITION, ?array $history = null)
     {
         $this->board = new Board();
         if (null !== $error = $this->load($fen ?? Board::DEFAULT_POSITION)) {
             throw new \InvalidArgumentException(\sprintf('Invalid fen: %s. Error: %s', $fen, $error));
         }
+        if (null !== $history) {
+            $this->history = $history;
```
Comment (akondas) [inline src/Chess.php:44]: maybe we can validate input with something like this:
```php
$this->history = array_map(fn(History $history): History => $history, $history);
```

**Review by akondas — APPROVED** [review]
LGTM :+1:

File: src/Chess.php (hunk 1 above)
Comment (garak) [reply src/Chess.php:44]: We can't use fn, unless we raise PHP requirements

**Pull request merged**